### PR TITLE
Fix lsp_servers[].file_patterns

### PR DESCRIPTION
**Pull request by wtfbbqhax**
## Description

This change fixes language detection when files are matched via lsp_servers[].file_patterns (for example **/*.h, **/*.c). serve() now uses an effective extension map that merges workspace extension mappings with extensions inferred from each LSP server’s file_patterns, so C/C++ headers and sources can resolve to the configured server language instead of falling back to plaintext (or wrong defaults).

### Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Refactoring (no functional changes)

## Related Issues

N/a

## Checklist

- [ ] I have read the CONTRIBUTING guide
- [x] My code follows the project's coding style
- [x] I have added tests that prove my fix/feature works
- [x] All new and existing tests pass
- [ ] I have updated the documentation accordingly
- [ ] I have updated the CHANGELOG.md (for user-facing changes)

## Additional Notes

- Added ServerConfig::build_effective_extension_map() and wired it into serve().
- Added tests for:
    - file-pattern-derived extension overrides (.c/.h -> cpp)
    - ignoring complex patterns that do not provide a simple extension
- Verified locally with: cargo test -p mcpls-core --lib (310 passed, 0 failed).

### CODE REVIEW

File: crates/mcpls-core/src/config/mod.rs [hunk 1]
```diff
@@ -116,6 +116,27 @@ impl WorkspaceConfig {
...
+
+    // Keep this conservative: only accept plain extension-like tokens.
+    if ext
+        .chars()
+        .all(|c| c.is_ascii_alphanumeric() || c == '_' || c == '-')
+    {
+        Some(ext.to_string())
+    } else {
+        None
+    }
```
Comment (bug-ops) [inline crates/mcpls-core/src/config/mod.rs:142]: Add direct unit tests for `extract_extension_from_pattern`

Currently this function is only tested indirectly through `test_build_effective_extension_map_overrides_with_file_patterns` and `test_build_effective_extension_map_ignores_complex_patterns_without_extension`.

Please add dedicated unit tests covering edge cases:

- Empty string pattern: `""` → `None`
- Pattern without dot: `"**/*"` → `None`
- Dotfile pattern: `".gitignore"` → `None`
- Multi-dot extensions: `"foo.tar.gz"` → `Some("gz")`

This ensures the conservative parsing logic is well-validated.

Optional: Consider returning `&str` instead of `String` since `rsplit_once` already returns slices of the input.

**Review by bug-ops — COMMENTED** [review]
Thanks for tackling this! A few items to polish before merge:

1. [test_serve_initializes_translator_with_extensions()](https://github.com/wtfbbqhax/mcpls/blob/1353ef986e3162dcc6d207b53d03d66c843a3575/crates/mcpls-core/src/bridge/translator.rs#L2755): This test calls the old API (workspace.build_extension_map()). Update it to test the new API (config.build_effective_extension_map()) or clarify if it's testing workspace-level mappings in isolation.
2. CHANGELOG.md: Document the default behavior change for C/C++ files.

File: crates/mcpls-core/src/config/mod.rs [hunk 2]
```diff
@@ -116,6 +116,32 @@ impl WorkspaceConfig {
...
+///
+/// Supports common patterns such as `**/*.rs` and `*.h`.
+/// Returns `None` for patterns without a simple trailing extension.
+fn extract_extension_from_pattern(pattern: &str) -> Option<String> {
+    let basename = pattern.rsplit('/').next().unwrap_or(pattern);
+    if basename.starts_with('.') {
+        return None;
+    }
+
+    let (_, ext) = pattern.rsplit_once('.')?;
```
Comment (bug-ops) [inline crates/mcpls-core/src/config/mod.rs]: `basename` is computed correctly above but then unused here - `rsplit_once` is called on the full pattern instead. This works in practice only because the symbol filter below rejects /, but the intent is clearly to operate on the filename part.

This also makes the dotfile guard and the split consistent: both now operate on the same string.

```suggestion
    let (_, ext) = basename.rsplit_once('.')?;
```

**Review by bug-ops — CHANGES_REQUESTED** [review]

**Comment by wtfbbqhax** [comment]
@bug-ops I'm slow to respond, so please feel free to slurp changes in without me 😅
You know your code, and how it should integrate better than I do

**Review by bug-ops — APPROVED** [review]
